### PR TITLE
[Docs] Updates Index/Home Page of Sol Docs / Adds "Getting Started"

### LINKED ISSUE #9475: [Docs] Add Getting Started Checklist to the "Basics" Section
As preparation for the upcoming Solidity Portal we decided to use the documentation as the "main source of truth". Right now, there is no part in the docs which summarizes easily how to "get started" as a newbie. This is mainly due to the fact that is isn't quite so easy to get started and we don't want to necessarily promote one tool over the other (or take other decisions for the user). To overcome this issue we could just implement a little checklist with links at the very beginning of the [basics section](https://solidity.readthedocs.io/en/v0.6.11/introduction-to-smart-contracts.html), which could cover stuff like (just rough idea, will obviously think more about the exact wording and content):

1. Get familiar with blockchain and the Ethereum / the EVM
2. Know what Smart Contracts are
3. Get the compiler
4. Choose an IDE (or tool)
5. "Try something easy"
6. [...]
7. [...]

#9175 for context on portal.

### PR DESCRIPTION
**Pull request by franzihei**
- restructures content on docs "home" page
- adds "getting started" section, which mostly contains what was there before, restructured and at parts slightly reworded

closes #9475

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -26,42 +26,58 @@ a 0.x version number `to indicate this fast pace of change <https://semver.org/#
...
+Getting Started
+---------------
+
+**1. Understand the Smart Contract Basics**
+
+If you are new to the concept of smart contracts we recommend you to get started by digging
+into the "Introduction to Smart Contracts" section, which covers:
+* :ref:`A simple example smart contract <simple-smart-contract>` written in Solidity.
+* Blockchain Basics.
+* The Ethereum Virtual Machine.
```
Comment (cameel) [inline docs/index.rst]: These used to be links. Or is the change on purpose?

```suggestion
* :ref:`A simple example smart contract <simple-smart-contract>` written in Solidity.
* :ref:`Blockchain Basics <blockchain-basics>`.
* :ref:`The Ethereum Virtual Machine <the-ethereum-virtual-machine>`.
```

File: docs/index.rst [hunk 2]
```diff
@@ -26,42 +26,58 @@ a 0.x version number `to indicate this fast pace of change <https://semver.org/#
...
-If you have any questions, you can try searching for answers or asking on the
-`Ethereum Stackexchange <https://ethereum.stackexchange.com/>`_, or
-our `gitter channel <https://gitter.im/ethereum/solidity/>`_.
+**4. Learn More**
 
-Ideas for improving Solidity or this documentation are always welcome,
-read our :doc:`contributors guide <contributing>` for more details.
+If you want to learn more about building decentralized applications on Ethereum, the
+`Ethereum Developer Resources <https://ethereum.org/en/developers/>`_
+can help you with further generic documentation around Ethereum, and a wide selection of tutorials,
```
Comment (cameel) [inline docs/index.rst]: Generic does not sound very enticing :)
```suggestion
can help you with further general documentation around Ethereum, and a wide selection of tutorials,
```

**Review by hrkrshnn — DISMISSED** [review]

**Comment by hrkrshnn** [comment]
Ah, please squash the commits.

**Review by cameel — DISMISSED** [review]

**Comment by chriseth** [comment]
Squashed.

**Review by hrkrshnn — APPROVED** [review]

**Pull request merged**